.circleci/config.yml

```diff
@@ -0,0 +1,184 @@
+# copied from https://github.com/circle-makotom/circle-advanced-setup-workflow/blob/main/.circleci/config.yml
```

Comment:
Do we need all of these? What about starting with minimal setup and adding things as we go, once they're required.